sklearn/utils/fixes.py

```diff
@@ -173,18 +173,6 @@ def __getstate__(self):
     from numpy.ma import MaskedArray    # noqa
 
 
-# Fix for behavior inconsistency on numpy.equal for object dtypes.
-# For numpy versions < 1.13, numpy.equal tests element-wise identity of objects
-# instead of equality. This fix returns the mask of NaNs in an array of
-# numerical or object values for all numpy versions.
-if np_version < (1, 13):
-    def _object_dtype_isnan(X):
-        return np.frompyfunc(lambda x: x != x, 1, 1)(X).astype(bool)
-else:
-    def _object_dtype_isnan(X):
-        return X != X
-
-
```

Comment:
The comment is misleading. The `numpy < 1.13` part only concerns the fact that `X != X` was not element-wise. However, `np.isnan` still does not work on object dtype.

So we still need
```py
def _object_dtype_isnan(X):
    return X != X
```